src/Asset/Capacity/AbstractCapacity.php

```diff
@@ -0,0 +1,207 @@
+<?php
+
+/**
+ * ---------------------------------------------------------------------
+ *
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ *
+ * http://glpi-project.org
+ *
+ * @copyright 2015-2023 Teclib' and contributors.
+ * @copyright 2003-2014 by the INDEPNET Development Team.
+ * @licence   https://www.gnu.org/licenses/gpl-3.0.html
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+namespace Glpi\Asset\Capacity;
+
+use DisplayPreference;
+use Glpi\Asset\Asset;
+use Log;
+use Profile;
+use ProfileRight;
+
+/**
+ * Abstract capacity that provides, among others, an empty implementation of some `\Glpi\Asset\Capacity\CapacityInterface`
+ * methods that can legitimately be effectless.
+ */
+abstract class AbstractCapacity implements CapacityInterface
+{
+    /**
+     * Constructor.
+     *
+     * Declared as final to ensure that constructor can be call without having to pass any parameter.
+     */
+    final public function __construct()
+    {
+    }
+
+    public function getSearchOptions(string $classname): array
+    {
+        return [];
+    }
+
+    public function onClassBootstrap(string $classname): void
+    {
+    }
+
+    public function onObjectInstanciation(Asset $object): void
+    {
+    }
+
+    public function onCapacityDisabled(string $classname): void
+    {
+    }
+
+    /**
+     * Delete logs related to relations between two itemtypes.
+     *
+     * @param string $source_itemtype
+     * @param string $linked_itemtype
+     * @return void
+     */
+    protected function deleteRelationLogs(string $source_itemtype, string $linked_itemtype): void
```

Comment:
If logs are not deleted when capacity is disabled, then the `History` tab will contain entries that are related to disabled capacities and to deleted data, and I think it can be confusing for end-user. If capacity is re-enabled data would not be restored, but log entries will still be present, and it wuld be even more confusing.

I made a test on my machine on a big DB (35M log entries). Removing all 600k lines related to `Monitor` itemtype using a similar query `` DELETE FROM `glpi_logs` WHERE `itemtype` = 'Monitor' OR `itemtype_link` = 'Monitor'; `` takes about 55 seconds. It is indeed too long, even if it is probably an edge case (it would be weird to disable a capacity that is so massively used that it produces so many log entries).

I added an entry in the TODO list in #15756 to try to find a solution for this.

Reply:
> My thoughts: I guess it would not be the end of the world if the logs are not immediately deleted and we wait for the cron to clean them.

I added an entry for this in the TODO list. It will be done later.